File: src/main/java/com/yerti/core/entity/CustomModel.java
package com.yerti.core.entity;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

public class CustomModel {

    private List<CustomModelPart> armorstands;


    public CustomModel() {
        this.armorstands = new ArrayList<>();
    }

    public List<CustomModelPart> getArmorstands() {
        return armorstands;
    }

    public void addArmorStand(CustomModelPart model) {
        armorstands.add(model);
    }

    public void removeArmorStand(CustomModelPart model) {
        Optional<CustomModelPart> optional = armorstands.stream().filter(m -> m.equals(model)).findFirst();

        optional.ifPresent(armorStandModel -> armorstands.remove(armorStandModel));
    }

    public void destroy() {
        for (CustomModelPart part : armorstands) {
            part.getArmorStand().remove();
        }
    }
}

File: src/main/java/com/yerti/core/entity/CustomModel.kt
package com.yerti.core.entity

import java.util.*

class CustomModel {
    private val armorstands: MutableList<CustomModelPart>
    fun getArmorstands(): List<CustomModelPart> {
        return armorstands
    }

    fun addArmorStand(model: CustomModelPart) {
        armorstands.add(model)
    }

    fun removeArmorStand(model: CustomModelPart) {
        val optional = armorstands.stream().filter { m: CustomModelPart -> m == model }.findFirst()
        optional.ifPresent { armorStandModel: CustomModelPart? -> armorstands.remove(armorStandModel) }
    }

    fun destroy() {
        for (part in armorstands) {
            part.armorStand.remove()
        }
    }

    init {
        armorstands = ArrayList()
    }
}